torch/distributed/_device_mesh.py

```diff
@@ -354,8 +354,8 @@ def get_dim_groups(
                 )
             return dim_groups
 
-    def size(self, dim: Optional[int] = None) -> int:
-        return self.mesh.numel() if dim is None else self.mesh.size(dim)
+    def size(self, mesh_dim: Optional[int] = None) -> int:
+        return self.mesh.numel() if mesh_dim is None else self.mesh.size(mesh_dim)
```

Comment:
do we want to add documentation in this PR or in follow up PR?

Reply:
> do we want to add documentation in this PR or in follow up PR?

I included all the documentation in the last one where we make device_mesh public.